examples/cfd/test_02_convection_nonlinear.ipynb

```diff
@@ -0,0 +1,354 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Example 2: Nonlinear convection in 2D\n",
+    "\n",
+    "Following the initial convection tutorial with a single steat variable $u$, we will now look at non-linear convection (step 6 in the original). This brings one new crucial challenge: computing a pair of coupled equations and thus updating two time-dependent variables $u$ and $v$.\n",
+    "\n",
+    "The full set of coupled equations is now\n",
+    "\n",
+    "$$\\frac{\\partial u}{\\partial t} + u \\frac{\\partial u}{\\partial x} + v \\frac{\\partial u}{\\partial y} = 0$$\n",
+    " \n",
+    "$$\\frac{\\partial v}{\\partial t} + u \\frac{\\partial v}{\\partial x} + v \\frac{\\partial v}{\\partial y} = 0$$\n",
+    "\n",
+    "and rearranging the discretised version gives us an expression for the update of both variables\n",
```

Comment:
This notation is confusing. Eventhiugh `u` and `v` appear in the equation it is not straightforward to see that there is a pde in u, specially considering both the second variable and the full field are called `v`